compiler/src/dotty/tools/dotc/transform/init/Semantic.scala

```diff
@@ -778,7 +776,20 @@ object Semantic {
     }
 
     /** Handle a new expression `new p.C` where `p` is abstracted by `value` */
-    def instantiate(klass: ClassSymbol, ctor: Symbol, args: List[ArgInfo]): Contextual[Value] = log("instantiating " + klass.show + ", value = " + value + ", args = " + args, printer, (_: Value).show) {
+    def instantiate(klass: ClassSymbol, ctor: Symbol, args: List[ArgInfo]): Contextual[Value] = log("instantiating " + klass.show + ", value = " + value + ", args = " + args.map(_.value.show), printer, (_: Value).show) {
+      def tryLeak(warm: Warm, argValues: List[Value]): Contextual[Value] =
+        val argInfos2 = args.zip(argValues).map { (argInfo, v) => argInfo.copy(value = v) }
+        val errors = Reporter.stopEarly {
+          given Trace = Trace.empty
+          warm.callConstructor(ctor, argInfos2)
+        }
+        if errors.nonEmpty then
+          val error = UnsafeLeaking(trace.toVector, errors.head)
```

Comment:
I thought about reporting the number of uses of the leaked value (i.e. the number of errors) in the warning in addition to the first error. But that's probably not worth the reduction in efficiency (we'd have to not use `Reporter.stopEarly`). If there turns out to be demand for it, perhaps some potential future PR could add an option like `-explain` to report more errors.